decidim-assemblies/app/controllers/decidim/assemblies/admin/assembly_members_controller.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Assemblies
+    module Admin
+      # Controller that allows managing assembly members.
+      #
+      class AssemblyMembersController < Decidim::Assemblies::Admin::ApplicationController
+        include Concerns::AssemblyAdmin
+
+        def index
+          authorize! :read, Decidim::AssemblyMember
+
+          @query = params[:q]
+          @status = params[:status]
+
+          @assembly_members = Decidim::Assemblies::Admin::AssemblyMembers.for(collection, @query, @status).page(params[:page]).per(15)
+        end
+
+        def new
+          authorize! :create, Decidim::AssemblyMember
+          @form = form(AssemblyMemberForm).instance
+        end
+
+        def create
+          authorize! :create, Decidim::AssemblyMember
+          @form = form(AssemblyMemberForm).from_params(params)
+
+          CreateAssemblyMember.call(@form, current_user, current_assembly) do
+            on(:ok) do
+              flash[:notice] = I18n.t("assembly_members.create.success", scope: "decidim.admin")
+              redirect_to assembly_members_path(current_assembly)
+            end
+
+            on(:invalid) do
+              flash[:alert] = I18n.t("assembly_members.create.error", scope: "decidim.admin")
+              render :new
+            end
+          end
+        end
+
+        def edit
+          @assembly_member = collection.find(params[:id])
+          authorize! :update, @assembly_member
+          @form = form(AssemblyMemberForm).from_model(@assembly_member)
+        end
+
+        def update
+          @assembly_member = collection.find(params[:id])
+          authorize! :update, @assembly_member
+          @form = form(AssemblyMemberForm).from_params(params)
+
+          UpdateAssemblyMember.call(@form, @assembly_member) do
+            on(:ok) do
+              flash[:notice] = I18n.t("assembly_members.update.success", scope: "decidim.admin")
+              redirect_to assembly_members_path(current_assembly)
+            end
+
+            on(:invalid) do
+              flash.now[:alert] = I18n.t("assembly_members.update.error", scope: "decidim.admin")
+              render :edit
+            end
+          end
+        end
+
+        def destroy
+          @assembly_member = collection.find(params[:id])
+          authorize! :destroy, @assembly_member
+
+          DestroyAssemblyMember.call(@assembly_member, current_user) do
+            on(:ok) do
+              flash[:notice] = I18n.t("assembly_members.destroy.success", scope: "decidim.admin")
+              redirect_to assembly_members_path(current_assembly)
+            end
+          end
+        end
+
+        def users
+          respond_to do |format|
+            format.json do
+              render json: [] && return if params.fetch(:term, "").empty?
+
+              query = current_organization.users&.order(name: :asc)
```

Comment:
This code is very similar to another PR, maybe we should extract this?

Reply:
@oriolgual What do you propose? Extract it to a concern? It's a good fit for an API endpoint, but it has no authentication, right?

Reply:
Also, we would have to add the `api` as dependency for the `admin` ... 😕